common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -482,32 +465,37 @@ def ensure_user_information(
     # It is important that we always execute the entire pipeline. Even if
     # behavior appears correct without executing a step, it means important
     # invariants have been violated and future misbehavior is likely.
-    user_inactive = user and not user.is_active
-    user_unset = user is None
+    def dispatch_to_login():
+        """Redirects to the login page."""
+        return redirect(_create_redirect_url(AUTH_DISPATCH_URLS[AUTH_ENTRY_LOGIN], strategy))
 
-    dispatch_to_login = (
-        ((is_login or is_login_2) and (user_unset or user_inactive))
-        or
-        ((is_register or is_register_2) and user_inactive)
-    )
-    dispatch_to_register = (is_register or is_register_2) and user_unset
-    reject_api_request = is_api and (user_unset or user_inactive)
+    def dispatch_to_register():
+        """Redirects to the registration page."""
+        return redirect(_create_redirect_url(AUTH_DISPATCH_URLS[AUTH_ENTRY_REGISTER], strategy))
 
-    if reject_api_request:
-        # Content doesn't matter; we just want to exit the pipeline
-        return HttpResponseBadRequest()
+    user_inactive = user and not user.is_active
 
-    if is_dashboard or is_profile:
-        return
+    if auth_entry == AUTH_ENTRY_LOGIN_API:
+        if not user:
+            return HttpResponseBadRequest()
 
-    # If the user has a linked account, but has not yet activated
-    # we should send them to the login page. The login page
-    # will tell them that they need to activate their account.
-    if dispatch_to_login:
-        return redirect(_create_redirect_url(AUTH_DISPATCH_URLS[AUTH_ENTRY_LOGIN], strategy))
+    elif auth_entry == AUTH_ENTRY_REGISTER_API:
+        if not user:
+            return HttpResponseBadRequest()
+        return {'user': user}
```

Comment:
Also, it looks like this can be combined with the AUTH_ENTRY_LOGIN_API block above